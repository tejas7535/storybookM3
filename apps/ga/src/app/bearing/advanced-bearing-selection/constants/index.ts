export * from './bearing-selection-constants';
