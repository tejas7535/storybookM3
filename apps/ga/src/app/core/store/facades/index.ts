export * from './calculation-paramaters/calculation-parameters.facade';
export * from './settings.facade';
