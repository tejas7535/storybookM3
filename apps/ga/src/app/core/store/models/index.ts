export * from './bearing-selection-state.models';
export * from './calculation-parameters-state.models';
export * from './calculation-result-state.models';
export * from './router-state.models';
export * from './settings-state.models';
