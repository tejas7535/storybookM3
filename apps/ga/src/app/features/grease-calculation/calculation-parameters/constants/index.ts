export * from './calculation-parameters-options';
export * from './calculation-parameters-validators';
