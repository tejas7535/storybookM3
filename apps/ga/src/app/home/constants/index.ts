export * from './homepage-card.constants';
