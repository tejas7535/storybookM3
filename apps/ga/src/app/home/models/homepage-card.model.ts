export interface HomepageCard {
  mainTitle: string;
  subTitle?: string;
  imagePath?: string;
  routerPath?: string;
  externalLink?: string;
  utmParameters?: string;
  trackingId?: string;
}
