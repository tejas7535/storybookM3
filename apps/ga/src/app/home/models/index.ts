export * from './homepage-card.model';
