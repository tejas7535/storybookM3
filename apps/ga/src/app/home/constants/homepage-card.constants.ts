import { AppRoutePath } from '@ga/app-route-path.enum';
import { UTM_PARAMS_DEFAULT } from '@ga/shared/constants';

import { HomepageCard } from '../models';

export const homepageCards: HomepageCard[] = [
  {
    mainTitle: 'calculator.title.main',
    subTitle: 'calculator.title.sub',
    imagePath: 'calculator.svg',
    routerPath: AppRoutePath.GreaseCalculationPath,
    utmParameters: UTM_PARAMS_DEFAULT,
    trackingId: 'greaseCalculation',
  },
  {
    mainTitle: 'greases.title.main',
    subTitle: 'greases.title.sub',
    imagePath: 'greases.jpg',
    externalLink: 'greases.externalLink',
    utmParameters: UTM_PARAMS_DEFAULT,
    trackingId: 'sources',
  },
  {
    mainTitle: 'lubricators.title.main',
    imagePath: 'lubricators.jpg',
    externalLink: 'lubricators.externalLink',
    utmParameters: UTM_PARAMS_DEFAULT,
    trackingId: 'lubricatorsLink',
  },
  {
    mainTitle: 'maintenance.title.main',
    imagePath: 'maintenance.jpg',
    externalLink: 'maintenance.externalLink',
    utmParameters: UTM_PARAMS_DEFAULT,
    trackingId: 'maintenanceLink',
  },
  {
    mainTitle: 'optime.title.main',
    imagePath: 'optime.jpg',
    externalLink: 'optime.externalLink',
    utmParameters: UTM_PARAMS_DEFAULT,
    trackingId: 'optimeLink',
  },
  {
    mainTitle: 'catalog.title.main',
    imagePath: 'catalog.jpg',
    externalLink: 'catalog.externalLink',
    utmParameters: UTM_PARAMS_DEFAULT,
    trackingId: 'catalogLink',
  },
  {
    mainTitle: 'contact.title.main',
    imagePath: 'contact.jpg',
    externalLink: 'contact.externalLink',
    utmParameters: UTM_PARAMS_DEFAULT,
    trackingId: 'contactLink',
  },
  {
    mainTitle: 'faq.title.main',
    imagePath: 'faq.jpg',
    externalLink: 'faq.externalLink',
    utmParameters: UTM_PARAMS_DEFAULT,
    trackingId: 'faqLink',
  },
];
