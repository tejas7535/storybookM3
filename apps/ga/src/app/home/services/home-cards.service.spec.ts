import { waitForAsync } from '@angular/core/testing';

import { BehaviorSubject } from 'rxjs';

import { createServiceFactory, SpectatorService } from '@ngneat/spectator/jest';

import { ApplicationInsightsService } from '@schaeffler/application-insights';
import { provideTranslocoTestingModule } from '@schaeffler/transloco/testing';

import { SettingsFacade } from '@ga/core/store/facades';
import { ENV, getEnv } from '@ga/environments/environments.provider';
import { ViCalculatorService } from '@ga/features/vi-calculator/services/vi-calculator.service';
import { TRACKING_NAME_HOMECARD } from '@ga/shared/constants';
import { PartnerAffiliateCode, PartnerVersion } from '@ga/shared/models';

import { HomepageCard } from '../models';
import { HomeCardsService } from './home-cards.service';
import { HomeCardsTrackingIds } from './home-cards-tracking-ids.enum';

describe('HomeCardsService', () => {
  let spectator: SpectatorService<HomeCardsService>;
  let service: HomeCardsService;
  let applicationInsightsService: ApplicationInsightsService;

  const partnerSubject = new BehaviorSubject<PartnerVersion>(undefined);
  const createServiceWithProductionConfig = createServiceFactory({
    service: HomeCardsService,
    imports: [provideTranslocoTestingModule({ en: {} })],
    providers: [
      {
        provide: ApplicationInsightsService,
        useValue: {
          logEvent: jest.fn(),
        },
      },
      {
        provide: SettingsFacade,
        useValue: {
          partnerVersion$: partnerSubject.asObservable(),
        },
      },
      {
        provide: ENV,
        useValue: { ...getEnv(), production: true },
      },
      {
        provide: ViCalculatorService,
        useValue: {
          showViscosityIndexCalculator: jest.fn(),
        },
      },
    ],
  });

  describe('when in production settings', () => {
    beforeEach(() => {
      spectator = createServiceWithProductionConfig();
      service = spectator.inject(HomeCardsService);
      applicationInsightsService = spectator.inject(ApplicationInsightsService);
      Object.defineProperty(window, 'open', {
        value: jest.fn(),
      });
    });

    it('should be created', () => {
      expect(service).toBeTruthy();
    });

    describe('when getting home cards', () => {
      let result: HomepageCard[];

      beforeAll(async () => {
        result = service.homeCards();
      });

      it('should get home cards for production', waitForAsync(() => {
        expect(result).toMatchSnapshot();
      }));

      describe('when performing any get external link action', () => {
        it('should log and open url with parameters', waitForAsync(() => {
          result[1].cardAction();

          expect(applicationInsightsService.logEvent).toHaveBeenCalledWith(
            TRACKING_NAME_HOMECARD,
            {
              card: HomeCardsTrackingIds.Sources,
            }
          );

          expect(window.open).toHaveBeenCalledWith(
            'homepage.cards.greases.externalLink?utm_source=grease-app&utm_medium=app',
            '_blank'
          );
        }));
      });

      describe('when performing external link action without paramaters', () => {
        it('should log and open url without parameters', waitForAsync(() => {
          result[6].cardAction();

          expect(applicationInsightsService.logEvent).toHaveBeenCalledWith(
            TRACKING_NAME_HOMECARD,
            {
              card: HomeCardsTrackingIds.ContactLink,
            }
          );

          expect(window.open).toHaveBeenCalledWith(
            'homepage.cards.contact.externalLink',
            '_blank'
          );
        }));
      });

      describe('when performing viscosity index calculator action', () => {
        it('should show viscosity index calculator', waitForAsync(() => {
          Object.defineProperty(service, 'isProduction', { value: false });
          result = service.homeCards();

          const viService = spectator.inject(ViCalculatorService);

          result[0].cardAction();

          expect(viService.showViscosityIndexCalculator).toHaveBeenCalled();
        }));
      });
    });
  });
  describe('when in Schmeckthal partner version setting', () => {
    beforeEach(() => {
      partnerSubject.next(PartnerVersion.Schmeckthal);
      spectator = createServiceWithProductionConfig();
      service = spectator.inject(HomeCardsService);
      applicationInsightsService = spectator.inject(ApplicationInsightsService);
      Object.defineProperty(window, 'open', {
        value: jest.fn(),
      });
    });

    it('should be created', () => {
      expect(service).toBeTruthy();
    });

    describe('when getting home cards', () => {
      let result: HomepageCard[];

      beforeAll(async () => {
        result = service.homeCards();
      });

      it('should get home cards for production', waitForAsync(() => {
        expect(result).toMatchSnapshot();
      }));

      describe('when performing opening partner email template', () => {
        it('should log and open email template', waitForAsync(() => {
          result[6].cardAction();

          expect(applicationInsightsService.logEvent).toHaveBeenCalledWith(
            TRACKING_NAME_HOMECARD,
            {
              card: HomeCardsTrackingIds.ContactLink,
            }
          );

          expect(window.open).toHaveBeenCalledWith(
            'mailto:technik@schmeckthal-gruppe.de?subject=Schmeckthal Grease App Anfrage',
            '_blank'
          );
        }));
      });

      describe('when performing any action with an external link', () => {
        it('should open a link with affiliate code', waitForAsync(() => {
          const affiliateCode =
            PartnerAffiliateCode[PartnerVersion.Schmeckthal];
          result[1].cardAction();

          expect(applicationInsightsService.logEvent).toHaveBeenCalledWith(
            TRACKING_NAME_HOMECARD,
            {
              card: HomeCardsTrackingIds.Sources,
            }
          );

          expect(window.open).toHaveBeenCalledWith(
            `homepage.cards.greases.externalLink?utm_source=grease-app&utm_medium=app&${affiliateCode}`,
            '_blank'
          );
        }));
      });
    });
  });
});
