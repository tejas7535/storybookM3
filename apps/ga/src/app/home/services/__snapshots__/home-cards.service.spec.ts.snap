// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeCardsService when in Schmeckthal partner version setting when getting home cards should get home cards for production 1`] = `
Array [
  Object {
    "cardAction": [Function],
    "mainTitle": "homepage.cards.calculator.title.main",
    "subTitle": "homepage.cards.calculator.title.sub",
    "templateId": "calculatorLogo",
  },
  Object {
    "cardAction": [Function],
    "imagePath": "assets/images/homepage/cards/greases.jpg",
    "mainTitle": "homepage.cards.greases.title.main",
    "subTitle": "homepage.cards.greases.title.sub",
    "templateId": "imageCard",
  },
  Object {
    "cardAction": [Function],
    "imagePath": "assets/images/homepage/cards/lubricators.jpg",
    "mainTitle": "homepage.cards.lubricators.title.main",
    "templateId": "imageCard",
  },
  Object {
    "cardAction": [Function],
    "imagePath": "assets/images/homepage/cards/maintenance.jpg",
    "mainTitle": "homepage.cards.maintenance.title.main",
    "subTitle": "homepage.cards.maintenance.title.sub",
    "templateId": "imageCard",
  },
  Object {
    "cardAction": [Function],
    "imagePath": "assets/images/homepage/cards/optime.jpg",
    "mainTitle": "homepage.cards.optime.title.main",
    "templateId": "imageCard",
  },
  Object {
    "cardAction": [Function],
    "imagePath": "assets/images/homepage/cards/catalog.jpg",
    "mainTitle": "homepage.cards.catalog.title.main",
    "templateId": "imageCard",
  },
  Object {
    "cardAction": [Function],
    "imagePath": "assets/images/homepage/cards/contact.jpg",
    "mainTitle": "partnerVersion.schmeckthal-gruppe.cards.contact.title.main",
    "subTitle": "partnerVersion.schmeckthal-gruppe.cards.contact.title.sub",
    "templateId": "imageCard",
  },
  Object {
    "cardAction": [Function],
    "imagePath": "assets/images/homepage/cards/faq.jpg",
    "mainTitle": "homepage.cards.faq.title.main",
    "templateId": "imageCard",
  },
]
`;

exports[`HomeCardsService when in production settings when getting home cards should get home cards for production 1`] = `
Array [
  Object {
    "cardAction": [Function],
    "mainTitle": "homepage.cards.calculator.title.main",
    "subTitle": "homepage.cards.calculator.title.sub",
    "templateId": "calculatorLogo",
  },
  Object {
    "cardAction": [Function],
    "imagePath": "assets/images/homepage/cards/greases.jpg",
    "mainTitle": "homepage.cards.greases.title.main",
    "subTitle": "homepage.cards.greases.title.sub",
    "templateId": "imageCard",
  },
  Object {
    "cardAction": [Function],
    "imagePath": "assets/images/homepage/cards/lubricators.jpg",
    "mainTitle": "homepage.cards.lubricators.title.main",
    "templateId": "imageCard",
  },
  Object {
    "cardAction": [Function],
    "imagePath": "assets/images/homepage/cards/maintenance.jpg",
    "mainTitle": "homepage.cards.maintenance.title.main",
    "subTitle": "homepage.cards.maintenance.title.sub",
    "templateId": "imageCard",
  },
  Object {
    "cardAction": [Function],
    "imagePath": "assets/images/homepage/cards/optime.jpg",
    "mainTitle": "homepage.cards.optime.title.main",
    "templateId": "imageCard",
  },
  Object {
    "cardAction": [Function],
    "imagePath": "assets/images/homepage/cards/catalog.jpg",
    "mainTitle": "homepage.cards.catalog.title.main",
    "templateId": "imageCard",
  },
  Object {
    "cardAction": [Function],
    "imagePath": "assets/images/homepage/cards/contact.jpg",
    "mainTitle": "homepage.cards.contact.title.main",
    "subTitle": undefined,
    "templateId": "imageCard",
  },
  Object {
    "cardAction": [Function],
    "imagePath": "assets/images/homepage/cards/faq.jpg",
    "mainTitle": "homepage.cards.faq.title.main",
    "templateId": "imageCard",
  },
]
`;
